File: SuperiorPackGroup/VendorsXtraForm.vb
Imports DXDAL.SPGData

Public Class VendorsXtraForm

    Private Sub deleteRepositoryItemButtonEdit_ButtonClick(ByVal sender As System.Object, ByVal e As DevExpress.XtraEditors.Controls.ButtonPressedEventArgs) Handles deleteRepositoryItemButtonEdit.ButtonClick

        If MessageBox.Show("Are you sure you want to delete this Vendor?", "Delete Vendor", MessageBoxButtons.YesNo, MessageBoxIcon.Hand) = Windows.Forms.DialogResult.Yes Then
            If VendorBLL.VendorInUse(CInt(vendorsGridView.GetFocusedRowCellValue(colOid))) = False Then
                vendorsGridView.DeleteRow(vendorsGridView.FocusedRowHandle)
            Else
                MessageBox.Show(String.Format("This vendor is referenced in some records.{0}You can't delete it at this time.", vbCrLf), "Vendor In Use", MessageBoxButtons.OK, MessageBoxIcon.Hand)
            End If
        End If

    End Sub

    Private Sub vendorsGridView_ValidateRow(ByVal sender As System.Object, ByVal e As DevExpress.XtraGrid.Views.Base.ValidateRowEventArgs) Handles vendorsGridView.ValidateRow

        If vendorsGridView.GetFocusedRowCellValue(colVendorName) Is Nothing OrElse String.IsNullOrEmpty(vendorsGridView.GetFocusedRowCellValue(colVendorName).ToString) Then
            e.ErrorText = "You must provide a Vendor name."
            e.Valid = False
        Else
            e.ErrorText = Nothing
            e.Valid = True
        End If

    End Sub

    Private Sub VendorsXtraForm_Load(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.Load

        DevExpress.Xpo.XPBaseObject.AutoSaveOnEndEdit = True
        CheckPermissions()

    End Sub

    Public Function CheckPermissions() As Boolean

        Try
            Dim permission As SPG.UserPermissionsRow = New UserPermissionsBLL().GetUserPermissionsByUserName(My.Settings.UserName, "General Lists")(0)

            Select Case permission.PermissionLevel
                Case "FULL", "Full"
                    Me.vendorsGridView.OptionsBehavior.Editable = True
                    Me.vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom
                    Me.deleteGridColumn.Visible = True
                Case "ADD", "Add"
                    Me.vendorsGridView.OptionsBehavior.Editable = True
                    Me.vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom
                    Me.deleteGridColumn.Visible = False
                Case "NONE", "None", "VIEW ASSIGNED", "EDIT ASSIGNED"
                    MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand)
                    Me.Close()
                    Return False
                Case "VIEW", "View"
                    Me.vendorsGridView.OptionsBehavior.Editable = False
                    Me.vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None
                    Me.deleteGridColumn.Visible = False
                Case "EDIT", "Edit"
                    Me.vendorsGridView.OptionsBehavior.Editable = True
                    Me.vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None
                    Me.deleteGridColumn.Visible = False
                Case Else
                    MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand)
                    Me.Close()
                    Return False
            End Select
            Return True
        Catch ex As InvalidOperationException
            Return False
        End Try

    End Function

End Class

File: SuperiorPackGroup/VendorsXtraForm.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using DXDAL.SPGData;

namespace SuperiorPackGroup
{
	public partial class VendorsXtraForm
	{

		public VendorsXtraForm()
		{
			InitializeComponent();
		}

		private void deleteRepositoryItemButtonEdit_ButtonClick(System.Object sender, DevExpress.XtraEditors.Controls.ButtonPressedEventArgs e)
		{

			if (MessageBox.Show("Are you sure you want to delete this Vendor?", "Delete Vendor", MessageBoxButtons.YesNo, MessageBoxIcon.Hand) == DialogResult.Yes)
			{
				if (VendorBLL.VendorInUse(Convert.ToInt32(vendorsGridView.GetFocusedRowCellValue(colOid))) == false)
				{
					vendorsGridView.DeleteRow(vendorsGridView.FocusedRowHandle);
				}
				else
				{
					MessageBox.Show(string.Format("This vendor is referenced in some records.{0}You can't delete it at this time.", Environment.NewLine), "Vendor In Use", MessageBoxButtons.OK, MessageBoxIcon.Hand);
				}
			}

		}

		private void vendorsGridView_ValidateRow(System.Object sender, DevExpress.XtraGrid.Views.Base.ValidateRowEventArgs e)
		{

			if (vendorsGridView.GetFocusedRowCellValue(colVendorName) == null || string.IsNullOrEmpty(vendorsGridView.GetFocusedRowCellValue(colVendorName).ToString()))
			{
				e.ErrorText = "You must provide a Vendor name.";
				e.Valid = false;
			}
			else
			{
				e.ErrorText = null;
				e.Valid = true;
			}

		}

		private void VendorsXtraForm_Load(System.Object sender, EventArgs e)
		{

			DevExpress.Xpo.XPBaseObject.AutoSaveOnEndEdit = true;
			CheckPermissions();

		}

		public bool CheckPermissions()
		{

			try
			{
				SPG.UserPermissionsRow permission = (new UserPermissionsBLL()).GetUserPermissionsByUserName(Properties.Settings.Default.UserName, "General Lists")[0];

				switch (permission.PermissionLevel)
				{
					case "FULL":
					case "Full":
						vendorsGridView.OptionsBehavior.Editable = true;
						vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom;
						deleteGridColumn.Visible = true;
						break;
					case "ADD":
					case "Add":
						vendorsGridView.OptionsBehavior.Editable = true;
						vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom;
						deleteGridColumn.Visible = false;
						break;
					case "NONE":
					case "None":
					case "VIEW ASSIGNED":
					case "EDIT ASSIGNED":
						MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand);
						Close();
						return false;
					case "VIEW":
					case "View":
						vendorsGridView.OptionsBehavior.Editable = false;
						vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None;
						deleteGridColumn.Visible = false;
						break;
					case "EDIT":
					case "Edit":
						vendorsGridView.OptionsBehavior.Editable = true;
						vendorsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None;
						deleteGridColumn.Visible = false;
						break;
					default:
						MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand);
						Close();
						return false;
				}
				return true;
			}
			catch (InvalidOperationException ex)
			{
				return false;
			}

		}

	}
}